src/corelibs/U2View/src/ov_msa/McaEditor.cpp

```diff
@@ -333,4 +338,14 @@ void McaEditor::addEditMenu(QMenu *menu) {
     editMenu->addAction(ui->getRedoAction());
 }
 
+void McaEditor::sl_gotoSelectedRead() {
+    MaEditorSelection selection = getSelection();
+    int rowIndex = selection.y();
+    CHECK(selection.height() > 0 && rowIndex >= 0 && rowIndex < maObject->getNumRows(), );
+
+    MultipleChromatogramAlignmentRow mcaRow = getMaObject()->getMcaRow(rowIndex);
+    qint64 rowStartPos = mcaRow->isComplemented() ? mcaRow->getCoreEnd() : mcaRow->getCoreStart();
```

Comment:
Both "getCoreEnd" and "getCoreStart" return "int". But some functions that are called inside these methods return qint64. Neither MCA Editor, nor MSA Editor could correctly process very long rows, so I offer to use "int" for the base positions.